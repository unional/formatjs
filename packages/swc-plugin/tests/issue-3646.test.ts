import {join} from 'path'
import {compile, FIXTURES_DIR} from './utils'

test.only('github issue #3646', async function () {
  const output = await compile(join(FIXTURES_DIR, `issue-3646.tsx`), {})
  expect(output.msgs).toMatchInlineSnapshot(`
    Array [
      Object {
        "defaultMessage": "Hello World!",
        "description": "The default message",
        "id": "GCdcoTfS3U",
      },
      Object {
        "defaultMessage": "Hello Nurse!",
        "description": "Another message",
        "id": "foo.bar.biff",
      },
    ]
  `)
  expect(output.code).toMatchSnapshot()
})
