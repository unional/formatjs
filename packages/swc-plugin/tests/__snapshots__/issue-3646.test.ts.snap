// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github issue #3646 2`] = `
"function _defineProperty(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
var _obj, _obj1;
var msgs = defineMessages({
    header: (_obj = {
        id: \\"GCdcoTfS3U\\",
        defaultMessage: \\"Hello World!\\"
    }, _defineProperty(_obj, \\"defaultMessage\\", \\"Hello World!\\"), _defineProperty(_obj, \\"description\\", \\"The default message\\"), _obj),
    content: (_obj1 = {
        id: \\"foo.bar.biff\\",
        defaultMessage: \\"Hello Nurse!\\"
    }, _defineProperty(_obj1, \\"id\\", \\"foo.bar.biff\\"), _defineProperty(_obj1, \\"defaultMessage\\", \\"Hello Nurse!\\"), _defineProperty(_obj1, \\"description\\", \\"Another message\\"), _obj1)
});
"
`;
